content/posts/0020-manually-testing-changes-to-paketo-buildpacks.md

```diff
@@ -0,0 +1,249 @@
+---
+title: Manually testing changes to a Paketo Buildpack
+date: "2023-05-23"
+slug: manually-testing-changes-to-paketo-buildpacks
+author: anthonydahanne
+---
+
+*While this article is mainly about how to manually test Java family Paketo buildpacks, still, most of the instructions should work fine with other Paketo buildpacks; and even non Paketo buildpacks!*
+
+Suppose that you started working on a new feature for the [`spring-boot`](https://github.com/paketo-buildpacks/spring-boot/) buildpack...
+
+## Unit tests first 
+
+Of course, during your development, you will make sure that you did not break any tests, regularly running `go test -v ./...` but you would also add new tests covering your new code; the classic combination `go test -v ./... -coverprofile=coverage.out` followed with `go tool cover -html=coverage.out` will allow you to verify you did not forget to test some errors or edge cases.
+
```

Comment:
ah! richgo, at first I wasn't sure why we used that, since to me the output palatability is not really enhanced.

But, ok, why not, maybe it made sense at one point.

And then, I read its [creator notice](https://github.com/kyoh86/richgo#notice-what-i-think-about-richgo) and I started envisioning removing it from everywhere...

All in all, I'd prefer not to mention it - it's 1 more library to know (hurdle for newcomers), not much value (go devs. are used to default tests output I assume), and not future proof at all (the notice)

Reply:
I need color to find anything in the `go test` output. That's a pain, that it's not being maintained.

Getting off on a tangent, but I'd support removing that since it's not maintained and we don't really need color in CI.